google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -369,38 +372,47 @@ def publish(self, topic, data, ordering_key="", **attrs):
                 "be sent as text strings."
             )
 
-        # Create the Pub/Sub message object.
-        message = types.PubsubMessage(
-            data=data, ordering_key=ordering_key, attributes=attrs
-        )
+        span_name = "{} publisher".format(topic)
+        span_attributes = {"data": data.decode()}
+        with create_span(span_name, attributes=span_attributes) as span:
+            if span is not None:
+                # Add the context of the span as an attribute
+                attrs["googclient_OpenTelemetrySpanContext"] = json.dumps(
```

Comment:
Sanity check - what if somebody explicitly sets a message attribute named `googclient_OpenTelemetrySpanContext`?

It's probably quite unlikely, but it probably makes sense to at least issue a warning that this message attribute will be overridden.